features/support/hooks.rb

```diff
@@ -8,17 +8,45 @@
   else
     begin
       flush_and_check_errors
-    rescue Capybara::Chromedriver::Logger::JsError => e
+    rescue LogMessage::JsError => e
       puts "Detected JS error, but ignored it. #{e}"
     end
   end
 end
 
 def flush_and_check_errors
-  # Do this manually rather than using `after_example!` so we can configure
-  # the log destination to be $stderr rather than $stdout. This prevents
-  # unformatted JavaScript errors from making their way into the Smokey loop's
-  # JSON output file and breaking it.
-  collector = Capybara::Chromedriver::Logger::Collector.new(log_destination: $stderr)
-  collector.flush_and_check_errors!
+  errors = extract_errors
+  return if errors.empty?
+
+  formatted_errors = errors.map(&:to_s)
+  error_list = formatted_errors.join("\n")
+
+  raise LogMessage::JsError, "Got some JS errors during testing:\n\n#{error_list}"
+end
+
+def extract_errors
+  errors = []
+
+  logs(:browser).each do |log|
+    message = LogMessage.new(log)
+    next if should_filter?(message)
+
+    errors << message if message.level == 'SEVERE'
+    $stderr.puts(message.to_s)
+  end
+
+  errors
+end
+
+def should_filter?(message)
+  %i[debug info warning].include?(message.level.downcase.to_sym)
+end
+
+def logs(type)
```

Comment:
```suggestion
def browser_logs(type)
```

This is more precise in the general context of the repo.